platform/pensando/sonic-platform-modules-dpu/dpu/utils/dpu_pensando_util.py

```diff
@@ -0,0 +1,172 @@
+# {C} Copyright 2023 AMD Systems Inc. All rights reserved
+
+import os
+import subprocess
+import sys
+import logging
+import time
+import re
+import json
+from sonic_platform.helper import APIHelper
+from sonic_py_common import syslogger
+
+QSFP_STAT_CTRL_CPLD_ADDR = "0x2"
+apiHelper = APIHelper()
+
+SYSLOG_IDENTIFIER = 'dpu_pensando_util'
+logger_instance = syslogger.SysLogger(SYSLOG_IDENTIFIER)
+
+def log_info(msg, also_print_to_console=False):
+    logger_instance.log_info(msg, also_print_to_console)
+
+def log_err(msg, also_print_to_console=True):
+    logger_instance.log_error(msg, also_print_to_console)
+
+def run_cmd(cmd):
+    status, output = subprocess.getstatusoutput(cmd)
+    if status != 0:
+        try:
+            log_err(cmd + " failed with err " + str(status) + "\n")
+        except:
+            log_err("platform_init: Failed file ops while logging")
+    else:
+        log_info(f"command executed : {cmd}")
+        log_info(output)
+    return output
+
+def fetch_dpu_files():
+    docker_id = apiHelper.get_dpu_docker_imageID()
+    cmd = "sudo docker cp {}:/tmp/fru.json /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/nic/bin/cpldapp /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/nic/lib/libpal.so /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/nic/lib/liblogger.so /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/nic/etc/VERSION.json /home/admin".format(docker_id)
+    run_cmd(cmd)
+    cmd = "sudo docker cp {}:/usr/bin/mmc /usr/local/bin".format(docker_id)
+    run_cmd(cmd)
```

Comment:
The logic of this is odd. There are files that are in the dpu docker, but are needed on the host. Ideally, either the files should just be present on the host and bind-mounted into the docker, or the host and container logic needs to be separated, or (in the case of the `mmc` binary) there should be a wrapper script on the host that calls the binary in the docker. In addition, the `admin` user shouldn't be used as a dumping ground like this; depending on build options, there's no guarantee that an `admin` user even exists.

This probably doesn't need to be handled immediately, but in the long term, this should be handled.

Reply:
Agreed @saiarcot895, will list this action item for phase-II.